### PR TITLE
Document persistentvolume gid annotation

### PR DESCRIPTION
**Pull request by wongma7**
Picking up #520 for feature kubernetes/kubernetes#29119

### CODE REVIEW

**Comment by pwittrock** [comment]
cc @pmorie

File: docs/user-guide/persistent-volumes/walkthrough.md [hunk 1]
```diff
@@ -36,6 +36,23 @@ NAME      LABELS       CAPACITY      ACCESSMODES   STATUS      CLAIM     REASON
 pv0001    type=local   10737418240   RWO           Available 
 ```
 
+### Access Control
+Storage configured with GID will only allow writing by pods using the same GID. Mismatched or missing GIDs will cause `permission denied` errors. Adding a GID to a `PersistentVolume` via the `pv.beta.kubernetes.io/gid` annotation allows `kubelet` to automatically add the GID to the pod that requires it. No coordination between an admin and end user is required. Setting the GID automatically is safe because this is only done for `PersistentVolumes` which are accessed through a `PersistentVolumeClaim`.
```
Comment (pmorie) [inline docs/user-guide/persistent-volumes/walkthrough.md]: wrap lines at 80 chars please, makes it much easier to comment on docs

Comment (wongma7) [inline docs/user-guide/persistent-volumes/walkthrough.md]: okay, done

**Comment by kelseyhightower** [comment]
@pmorie Thanks for the initial review. Can you PTAL now that the lines are wrapped?

File: docs/user-guide/persistent-volumes/walkthrough.md [hunk 2]
```diff
@@ -36,6 +36,31 @@ NAME      LABELS       CAPACITY      ACCESSMODES   STATUS      CLAIM     REASON
 pv0001    type=local   10737418240   RWO           Available 
 ```
 
+### Access Control
+Storage configured with GID will only allow writing by pods using the same GID.
+Mismatched or missing GIDs will cause `permission denied` errors. Adding a GID
+to a `PersistentVolume` via the `pv.beta.kubernetes.io/gid` annotation allows
```
Comment (pmorie) [inline docs/user-guide/persistent-volumes/walkthrough.md]: I would phrase this as 'annotating a `PersistentVolume` with a GID'.  The gid isn't really added to the volume, per-se.

File: docs/user-guide/persistent-volumes/walkthrough.md [hunk 3]
```diff
@@ -36,6 +36,31 @@ NAME      LABELS       CAPACITY      ACCESSMODES   STATUS      CLAIM     REASON
...
+`kubelet` to automatically add the GID to the pod that requires it. No
+coordination between an admin and end user is required. Setting the GID
+automatically is safe because this is only done for `PersistentVolumes` which
```
Comment (pmorie) [inline docs/user-guide/persistent-volumes/walkthrough.md]: I'm not sure what 'safe' is supposed to mean here.

File: docs/user-guide/persistent-volumes/walkthrough.md [hunk 4]
```diff
@@ -36,6 +36,31 @@ NAME      LABELS       CAPACITY      ACCESSMODES   STATUS      CLAIM     REASON
...
+```yaml
+kind: PersistentVolume
+apiVersion: v1
+metadata:
+  name: pv1
+  annotations:
+    pv.beta.kubernetes.io/gid: "1234"
+#...
+```
+
```
Comment (pmorie) [inline docs/user-guide/persistent-volumes/walkthrough.md]: @wongma7 Add an explanation of what this means for the pod.  Make sure to note that currently, the GIDs associated with PVs a pod consumes will not be present on the pod resource itself.  This reminds me, I need to dig up the issue for security context status and update it with this case.

**Comment by kelseyhightower** [comment]
@wongma7 Seems like you're getting close. Thanks for your awesome on documenting this area of Kubernetes.

File: docs/user-guide/persistent-volumes/walkthrough.md (hunk 3 above)
Comment (wongma7) [inline docs/user-guide/persistent-volumes/walkthrough.md]: I'm not sure either and I don't think the reader needs this reassurance(?), I will remove this

**Comment by kelseyhightower** [comment]
LGTM. This address the feedback.

**Pull request merged**